### ISSUE TITLE
Support same variable name across different input sources

### ISSUE THREAD

**Issue by juancq** [mmcdermott/EventStreamGPT#53]
I have the following input sources:

```yaml
inputs: 
  subjects:
    ...
  admissions:
    input_df: 'admissions.parquet'
    ...
  emergency:
    input_df: 'emergency.parquet'
    ...

measurements: 
  ...
  dynamic:
    multi_label_classification: 
      admissions:
        - facility_type
        - procedure
        ...
      emergency:
        - facility_type
        - triage_category
        ...
```

Getting the following error when running my build_dataset script: 
```bash
ValueError: facility_type differs across input sources. 
```

How can I have input sources with the same name (facility_type in admissions and in emergency)? Or how can I implement aliasing? i.e. something along the lines of: 

```yaml
...
measurements: 
  ...
  dynamic:
    multi_label_classification: 
      admissions:
        - facility_type
        - procedure
        ...
      emergency:
        - facility_type as ed_facility_type
        - triage_category
        ...
```

**Comment by mmcdermott** [comment]
@juancq this is a great issue. What would you like the desired behavior to be here in the case of collisions? Is it acceptable if the default behavior as of now is to keep both records even in the case of matching timestamps? E.g., if admissions has a facility type of X and emergency of Y both at 12/1/1990 11:30 a.m., then the resulting dataset would have records of facility type measurements of both X and Y in the event at that timestamp (and the event at that timestamp would have an event type containing at least `ADMISSION&EMERGENCY`, presuming default event types)? Or would you be expecting / want some aggregation or de-duplication?

**Comment by mmcdermott** [comment]
Sounds good @juancq. I'll plan to take a stab at this next week at some point. It won't be actually challenging, but it will require a slightly more explicit form of the polars reading in of the input data sources and merging them that can account for overlapping column names.

**Comment by mmcdermott** [comment]
@juancq, I started trying to craft a failing example of this in the synthetic tutorial, so I had something to try and fix, and too my surprise it actually seemed to work ok there. This is in this branch: https://github.com/mmcdermott/EventStreamGPT/compare/dev...enable_colliding_input_columns?expand=1

Can you expand a bit on your use case in terms of other details about the datasets so I can better understand what issue might be preventing your case from working that isn't causing an issue for the sample dataset?

**Comment by juancq** [comment]
It seems to be resolved now. I was using an older version of the build_dataset.py script. It was fixed in: 0ce4fed

**Comment by mmcdermott** [comment]
Huh, interesting. Glad to see it is working now! Closing this issue then.
